src/HalfIntegers.jl

```diff
@@ -58,7 +58,7 @@ HalfInteger(x::BigFloat) = BigHalfInt(x)
         if isinteger(x)
             Base.unsafe_rational(T, tx >> 1, one(tx))
         else
-            Base.unsafe_rational(T, tx, oftype(tx,2))
+            Base.unsafe_rational(T, promote(tx, T(2))...)
```

Comment:
```suggestion
            Base.unsafe_rational(T, tx, twice(one(tx)))
```
`promote` might unnecessarily widen `tx`, what do you think about this alternative proposal?

Reply:
Seems reasonable to me, I'll try this out